dandi/publish/views/draft_version.py

```diff
@@ -83,3 +88,36 @@ def draft_publish_view(request, dandiset__pk):
     dandiset.draft_version.lock(request.user)
     publish_version.delay(dandiset.id, request.user.id)
     return Response(status=status.HTTP_204_NO_CONTENT)
+
+
+@swagger_auto_schema(
+    method='POST',
+    request_body=UserSerializer(many=True),
+)
+@api_view(['POST'])
+@permission_required_or_403('owner', (DraftVersion, 'dandiset__pk', 'dandiset__pk'))
+@permission_classes([IsAuthenticatedOrReadOnly])
+def draft_owners_view(request, dandiset__pk):
+    draft_version = get_object_or_404(Dandiset, pk=dandiset__pk).draft_version
+
+    new_owners_serializer = UserSerializer(data=request.data, many=True)
+    if not new_owners_serializer.is_valid():
+        return Response(new_owners_serializer.errors, status=status.HTTP_400_BAD_REQUEST)
+    new_owners = [
+        get_object_or_404(User, username=owner['username'])
```

Comment:
If a single owner within the request isn't found, I'm not sure 404 is correct. 400 would make more sense to me, since we're talking about data within the request, not the ultimate resource target of the request.